airflow/kubernetes/kubernetes_helper_functions.py

```diff
@@ -37,7 +37,7 @@ def _strip_unsafe_kubernetes_special_chars(string: str) -> str:
     :param string: The requested Pod name
     :return: Pod name stripped of any unsafe characters
     """
-    return ''.join(ch.lower() for ch in list(string) if ch.isalnum())
+    return ''.join(ch.lower() for ch in list(string) if ch.isalnum()).encode('ascii', 'ignore').decode()
```

Comment:
```suggestion
    return string.encode('ascii', 'ignore').decode().lower()
```

I think this will be a bit more performant (less split and join happening)

Reply:
How about using unidecode insteead of stripping? https://stackoverflow.com/questions/517923/what-is-the-best-way-to-remove-accents-normalize-in-a-python-unicode-string

Reply:
Ah I see the comment, above. But I think it would be better to use unidecode. Maybe a little extreme but there are a few words in Polish for example, that contain only, or mostly accented characters. Not that they are often used, but this might lead to quite some ambiguity. And it might be worse in other languages.

Example words in Polish with only accented characters:

żółć, łóż, łżą, żąć, żął

Few short ones:

łódź, łażącą, łożącą, łóżmyż, łóżże, łżącą, żąłeś, żęłaś, żółcą, żółcę

Few longer words with mostly 

niedołężność, współdźwięcznością, żółtoróżowością, pięćdziesięciopięcioipółlatkąście

:D

Reply:
BTW. In case you had any doubts, Polish is one of the hardest languages in the world.

Here you can see about 1/3 of all the possible 'variants' of words that you can make from "play" = "gra". 

![Screenshot 2021-07-18 16 03 14](https://user-images.githubusercontent.com/595491/126070241-1f27188f-9eb5-4854-87f4-316d5c6e73d0.png)

Reply:
FWIW if we're going for a complex solution, Mozilla's [unicode-slugify](https://github.com/mozilla/unicode-slugify) normalises even more things, including characters from Chinese (my native language) 😁

Reply:
That would be even better! BTW. I once investigated the problem (turned out to be MySQL encoding problem) where someone used 🦅 in the dag_id 😱 . So called "Eagle problem".  I wonder how unicode-slugify would deal with that one :)

Reply:
Good enough:

![Screenshot 2021-07-18 17 12 13](https://user-images.githubusercontent.com/595491/126072419-1da3c9b4-1f34-437d-ad44-f0288ec60c63.png)

Reply:
@potiuk The issue its only for the pod name, do we really need to translate with a more complex solution? If so, i can make the changes.

Reply:
Well, I'd argue that it can't get less complex than calling a "slugify" method - I think trying to fix it manually when you have good and proven library to do it (stamped by mozilla) and even (as @uranusjr pointed out) handles Chinese characters is in many ways simpler solution.

Reply:
@potiuk Yeah, but we are adding a package dependency to airflow only for handling the complexity of translating chars in pod name which we already strip characters incompatible ascii chars like - or _.

Reply:
@potiuk anyways, its my opinion, if everybody agree adding the dependency unicode-uglify to airflow, i can make the changes in the PR...

Reply:
I see no problem whatsoever with adding a proven dependency that does a good job and has rather straightforward and expected dependencies - especially if it can save not only us (but also our current and future users) from even seeing similar errors. 

From the `product` point of view, if we can get such proven solution that saves us headeaches in the future is the right way. The case where POD has Chinese-only characters mentioned by @uranusjr WILL eventually happen, and when it will, we will have to do it anyway, so why not doing it now :).

```
unicode-slugify==0.1.3
  - six [required: Any, installed: 1.16.0]
  - unidecode [required: Any, installed: 1.2.0]
```

Reply:
I just realised unicode-slugify has a hard dependency on unidecode, which brings back the GPL issue. Since unicode-slugify is a pretty thin layer over unidecode anyway, it’s probably to hand-roll an implementation in Airflow instead. I’ll take some time to look into this tomorrow.

Reply:
Ah yeah. VERY GOOD CALL @uranusjr . I did not notice it's GPL. In this case we should definitely NOT bring it.